### PR TITLE
Format libcore with rustfmt

### PR DESCRIPTION
**Pull request by dtolnay**
I am interested in whether we can begin cautious incremental progress on #66688 and assess along the way whether we can keep the disruption sufficiently small.

This PR applies rustfmt with default settings to files in src/libcore *that are not involved in any currently open PR* to minimize merge conflicts. The list of files involved in open PRs was determined by querying GitHub's GraphQL API  [with this script](https://gist.github.com/dtolnay/aa9c34993dc051a4f344d1b10e4487e8).

With the list of files from the script in `outstanding_files`, the relevant commands were:

```console
$ find src/libcore -name '*.rs' | xargs rustfmt --edition=2018
$ rg libcore outstanding_files | xargs git checkout --
```

Repeating this process several months apart should get us coverage of most of the rest of libcore.

### CODE REVIEW

**Comment by dtolnay** [comment]
I need to update `tidy` to accept feature names on their own line. Will take care of that in a separate PR.

```rust
#[unstable(
    feature = "derive_clone_copy",
```

**Comment by petrochenkov** [comment]
>This PR applies rustfmt with default settings to files

"Default" here means https://github.com/rust-lang/rust/blob/master/rustfmt.toml or really default?

**Comment by petrochenkov** [comment]
cc https://github.com/rust-lang/rust/pull/65939

**Comment by dtolnay** [comment]
Ah, it's with that rustfmt.toml. It got picked up from running rustfmt in the same directory.

**Comment by sfackler** [comment]
r=me

**Comment by dtolnay** [comment]
Rebased on https://github.com/rust-lang/rust/pull/66717, removing from this PR 2 files that have been touched since first opening this PR:

- src/libcore/cmp.rs
- src/libcore/panicking.rs

and 6 files that appear in newly opened PRs:

- src/libcore/fmt/builders.rs
- src/libcore/fmt/num.rs
- src/libcore/hint.rs
- src/libcore/iter/traits/collect.rs
- src/libcore/num/f32.rs
- src/libcore/ops/try.rs

**Comment by dtolnay** [comment]
- Updated UI test in src/test/ui/macros/unknown-builtin.stderr, which includes a line from a builtin macro (https://github.com/rust-lang/rust/commit/166471e7f1f05fe272a4df99c20c1ffc0204a25f)
- Updated to pass `--edition=2018` to rustfmt or else it rewrites `$crate::r#try!($expr)` to `$crate::try!($expr)`
- Updated
https://github.com/rust-lang/rust/blob/809e180a76ce97340bf4354ff357bc59e3ca40b2/src/libcore/macros/mod.rs#L1296-L1297
to reorder `feature` before `soft` so that tidy sees `feature` on the line it expects

File: src/libcore/char/mod.rs [hunk 1]
```diff
@@ -45,13 +45,13 @@ use crate::fmt::{self, Write};
...
-const MAX_ONE_B: u32   =     0x80;
-const MAX_TWO_B: u32   =    0x800;
-const MAX_THREE_B: u32 =  0x10000;
+const TAG_CONT: u8 = 0b1000_0000;
+const TAG_TWO_B: u8 = 0b1100_0000;
+const TAG_THREE_B: u8 = 0b1110_0000;
+const TAG_FOUR_B: u8 = 0b1111_0000;
+const MAX_ONE_B: u32 = 0x80;
+const MAX_TWO_B: u32 = 0x800;
+const MAX_THREE_B: u32 = 0x10000;
```
Comment (RalfJung) [inline src/libcore/char/mod.rs]: Is there a way to register exceptions from rustfmt? The manual alignment here, at least for the first tags that are written in binary, was quite useful IMO.

Comment (dtolnay) [reply src/libcore/char/mod.rs]: There is, but I don't think the alignment is enough of a readability improvement to be worth making an exception here. I think the underscore grouping is sufficient to read them clearly. I am amenable in general to making rustfmt exceptions but I think it doesn't make enough of a difference in this case.

**Pull request merged**